### PR TITLE
[pr2eus] Add make-robot-interface-from-name function to create robot-interface instance from name

### PR DESCRIPTION
**Pull request by garaemon**
see https://github.com/jsk-ros-pkg/jsk_visualization/issues/227

It's important to write code independent from robot!

### CODE REVIEW

**Comment by snozawa** [comment]
To create each robot interface, we tipically write the following codes:
```
(defclass hrp2jsk-interface
  :super rtm-ros-robot-interface
  :slots ())
(defmethod hrp2jsk-interface
  (:init (&rest args)
         (send-super* :init :robot hrp2jsk-robot args)))

(defun hrp2jsk-init (&rest args)
  (if (not (boundp '*ri*))
      (setq *ri* (instance* hrp2jsk-interface :init args)))
  (if (not (boundp '*hrp2jsk*))
      (setq *hrp2jsk* (instance hrp2jsk-robot :init))))
```
But, I think this code can be replaced like this:
```
(defun hrp2jsk-init (&rest args)
  (if (not (boundp '*ri*))
      (setq *ri* (instance* robot-interface :init :robot hrp2jsk-robot args)))
  (if (not (boundp '*hrp2jsk*))
      (setq *hrp2jsk* (instance hrp2jsk-robot :init))))
```
In this case, we do not need to define each robot interface inheriting robot-interface class.

**Comment by garaemon** [comment]
You mean we do not need subclass of robot-interface?

**Comment by snozawa** [comment]
Sorry, robot dependent interface subclass is necessary...
So, please neglect the above comment:https://github.com/jsk-ros-pkg/jsk_pr2eus/pull/97#issuecomment-71825671

File: pr2eus/robot-interface.l [hunk 1]
```diff
@@ -892,4 +892,15 @@
           (nreverse tms))
     ))
 
+(defun make-robot-interface-from-name (name &rest args)
+  "make a robot model from string: (make-robot-model \"pr2\")"
+  (let ((name (string-downcase name)))
+    (let ((klasses (flatten (send robot-interface :hierarchy))))
+      (dolist (klass klasses)
+        (let ((klass-name (string-downcase (send klass :name))))
+        (if (or (string= klass-name name)
```
Comment (snozawa) [inline pr2eus/robot-interface.l]: I'd like to keep searching rules simple.
Why isn't it sufficient to search [robot-name]-interface?
How about to define function like this?
```
(let ((klass (read-from-string (format nil "~A-interface" robot-name))))
  (if (and (boundp klass) (subclassp (eval klass) robot-interface))
      (instance* (eval klass) :init args)
    (error "No such subclass of robot-interface : ~A~%" klass)))
```

Comment (garaemon) [inline pr2eus/robot-interface.l]: looks good!

**Comment by garaemon** [comment]
updated

**Comment by k-okada** [comment]
why https://github.com/euslisp/jskeus/blob/master/irteus/irtutil.l#L378   is not used in here?
can we remove `make-robot-model-from-name` ?

**Pull request merged**